text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+possibility of [typosquatting](https://en.wikipedia.org/wiki/Typosquatting).
+Allowed characters are [the same as for a valid hostname](https://en.wikipedia.org/wiki/Hostname#Syntax).
+
+The directory structure will contain a folder for each dot-separated segment of
+the dependency’s organization name and in the lowest level directory there will
+be a file named after the dependency name with the extension `toml`, this is
+because the metadata file will be TOML format. For example, with
+`com.example.dep-a`, there would be the following folder structure:
+
+```
+com
+└── example
+    └── dep-a.toml
+```
+
+As mentioned previously, each individual metadata file has a file name
+consisting of the dependency name with an extension of `toml`. The internal
+format consists of a single table called `versions` which is an array of tables
+containing all of the versions for that particular dependency. Each version
+entry requires a `uri`, `version`, `checksum`, `arch`, `os`, and `license`. It
+may also have `name`, `purl`, `strip-components`, and `cpes` although these are
+optional. The `cpes` entry is an array of strings identifying all of the CPEs
+for that dependency. The `license` is itself an array of tables containing
+`type` and `uri` of the license for the dependency.
+
+For example:
+```toml
+[[versions]]
+cpes = [ "cpe:2.3:a:apache:maven:3.8.6:*:*:*:*:*:*:*" ]
+name = "Apache Maven"
+purl = "pkg:generic/apache-maven@3.8.6"
+checksum = "sha256:c7047a48deb626abf26f71ab3643d296db9b1e67f1faa7d988637deac876b5a9"
+arch = "x86_64"
+os = "linux"
+distro = "ubuntu-18.04"
+uri = "https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.8.6/apache-maven-3.8.6-bin.tar.gz"
+version = "3.8.6"
+strip-components = 1
+
+  [[versions.licenses]]
+  type = "Apache-2.0"
+  uri = "https://www.apache.org/licenses/"
+
+[[versions]]
+cpes = [ "cpe:2.3:a:apache:mvnd:0.7.1:*:*:*:*:*:*:*" ]
+name = "Apache Maven Daemon"
+purl = "pkg:generic/apache-mvnd@0.7.1"
+chekcsum = "sha256:ac0b276d4d7472d042ddaf3ad46170e5fcb9350981af91af6c5c13e602a07393"
+arch = "x86_64"
+os = "linux"
+uri = "https://github.com/apache/maven-mvnd/releases/download/0.7.1/mvnd-0.7.1-linux-amd64.zip"
+version = "0.7.1"
+
+  [[versions.licenses]]
+  type = "Apache-2.0"
+  uri = "https://www.apache.org/licenses/"
+```
+## Dependency Validation
+In `buildpack.toml` we will add a section to metadata for specifying dependency
+validation parameters. This is a way that buildpacks can state that they do or
+do not support certain versions of dependencies.
+
+A buildpack does not need to include this section, it is optional. If included,
+the buildpack and libraries like `libpak` and `packit` may use the information
+to fail if dependency versions are requested by a user that might cause
+problems for the buildpack. It is the buildpacks responsibility to process the
+validations and react to them, whether that be warning the user or even
+failing.
```

Comment:
Is that to say `libpak` and `packit` could provide a `validate` function, but buildpacks are responsible to call it, iterate over the result and decide on log output and success vs. failure?

Reply:
Because this `validation` section is staying in the `metadata` section of `buildpack.toml` it will continue to be Paketo specific. Therefore Paketo libraries would be responsible for respecting this metadata but it will not become an explicit part of the outward facing interface of the decouple dependency metadata (although users creating their own buildpacks may want to leverage that Paketo specific API)